src/listmode_buildblock/CListModeDataSAFIR.cxx

```diff
@@ -0,0 +1,134 @@
+/* CListModeDataSAFIR.cxx
+
+Coincidence LM Data Class for SAFIR: Implementation
+Jannis Fischer
+jannis.fischer@cern.ch
+
+	Copyright 2015 ETH Zurich, Institute of Particle Physics
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+#include <iostream>
+#include <fstream>
+#include <typeinfo>
+
+#include "stir/ExamInfo.h"
+#include "stir/Succeeded.h"
+#include "stir/is_null_ptr.h"
+
+#include "boost/static_assert.hpp"
+#include "boost/pointer_cast.hpp"
+
+#include "stir/listmode/CListModeDataSAFIR.h"
+
+
+//#include "stir/ExamInfo.h"
+
+
+#ifndef STIR_NO_NAMESPACES
+using std::cerr;
+using std::endl;
+using std::ios;
+using std::fstream;
+using std::ifstream;
+using std::istream;
+#endif
+
+START_NAMESPACE_STIR;
+
+
+template <class CListRecordT>
+CListModeDataSAFIR<CListRecordT>::
+CListModeDataSAFIR(const std::string& listmode_filename, const std::string& crystal_map_filename, const std::string& template_proj_data_filename)
+  : listmode_filename(listmode_filename), map(boost::make_shared<ListEventRecordMapFromFile>(crystal_map_filename))
+{
+	this->exam_info_sptr.reset(new ExamInfo);
+
+	// Here we are reading the scanner data from the template projdata
+	shared_ptr<ProjData> template_proj_data_ptr =
+//			ProjData::read_from_file("muppet.hs");
+			ProjData::read_from_file(template_proj_data_filename);
+	scanner_sptr = shared_ptr<Scanner>( new Scanner(*template_proj_data_ptr->get_proj_data_info_ptr()->get_scanner_ptr()));
+
+	if( open_lm_file() == Succeeded::no )
+	{
+		error("CListModeDataSAFIR: Could not open listmode file %s\n", listmode_filename.c_str());
```

Comment:
Since it is only string concatenation I used error("..." + listmode_filename + "\n"). would that work for you? With the post-processing it should anyhow complain earlier.